CMake/curl-config.cmake

```diff
@@ -9,6 +9,11 @@ if(NOT CURL_FIND_COMPONENTS)
     endif()
 endif()
 
+include(CMakeFindDependencyMacro)
+if(CURL_FIND_REQUIRED_libcurl)
+    find_dependency(OpenSSL "@OPENSSL_VERSION_MAJOR@")
+endif()
```

Comment:
Once content substitution is done then the input file should be renamed with a `.in` extension so that nothing ever tries to read it as a valid `.cmake` file.

Reply:
Done. + rebased since curl-config.cmake also changed upstream from original PR.

Reply:
Thanks.

The `CURL_FIND_REQUIRED_libcurl` value is set only if `find_package` is called with `REQUIRED`.  Even if it is not a required package we still need to look for curl's dependencies.